### PR TITLE
Update for latest syntax changes

### PR DESCRIPTION
**Pull request by Shairam**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -163,107 +160,129 @@ service<http:Service> StudentData bind studentServiceListener {
         path: "/addStudent"
     }
     // Add Students resource used to add student records to the system.
-    addStudents(endpoint httpConnection, http:Request request) {
+    resource function addStudents(http:Caller httpConnection, http:Request request) {
```
Comment (Anoukh) [inline README.md]:
```suggestion
    resource function addStudents(http:Caller caller, http:Request request) {
```
Shall we use the variable name `caller` ?

File: README.md [hunk 2]
```diff
@@ -163,107 +160,129 @@ service<http:Service> StudentData bind studentServiceListener {
...
+        studentData.requestCounts += 1;
+        http:Response response = new;
 
         // Accepting the JSON payload sent from a request.
-        json payloadJson = check request.getJsonPayload();
+        json|error payloadJson =  request.getJsonPayload();
 
-        //Converting the payload to Student type.
-        Student studentDetails = check <Student>payloadJson;
+        if(payloadJson is json){
```
Comment (Anoukh) [inline README.md]:
```suggestion
        if (payloadJson is json) {
```

File: README.md [hunk 3]
```diff
@@ -163,107 +160,129 @@ service<http:Service> StudentData bind studentServiceListener {
...
+           Student|error studentDetails = Student.convert(payloadJson);
 
-        // Send the response back to the client with the returned JSON value from insertData function.
-        response.setJsonPayload(returnValue);
-        _ = httpConnection->respond(response) but { error e => log:printError("Error sending response", err = e)};
+            if (studentDetails is Student) {
+                io:println(studentDetails);
+                // Calling the function insertData to update database.
+                json returnValue = insertData( untaint studentDetails.name,  untaint studentDetails.age, untaint studentDetails.mobNo, untaint studentDetails.
+                    address);
```
Comment (Anoukh) [inline README.md]: Shall we reformat this?

File: README.md [hunk 4]
```diff
@@ -163,107 +160,129 @@ service<http:Service> StudentData bind studentServiceListener {
...
 
-        // Match operator used to check if the response returned value with one of the types below.
-        match returnValue {
-            table tableReturned => dataTable = tableReturned;
-            error e => io:println("Select data from student table failed: "
-                    + e.message);
+        if (returnValue is error) {
+            io:println("Select data from student table failed");
+        }
+        else {
```
Comment (Anoukh) [inline README.md]: Can be moved to the top line

Comment (Anoukh) [reply README.md]: Check in all places

File: README.md [hunk 5]
```diff
@@ -163,107 +160,129 @@ service<http:Service> StudentData bind studentServiceListener {
...
         // Sending back the converted JSON data to the request made to this service.
         response.setJsonPayload(untaint status);
-        _ = httpConnection->respond(response) but { error e => log:printError("Error sending response", err = e) };
+        var result = httpConnection->respond(response);
+        if (result is error) {
+            log:printError("Error sending response", err = result);
+        }
 
-        _ = observe:finishSpan(childSpanId);
+        if (childSpanId is int ) {
```
Comment (Anoukh) [inline README.md]:
```suggestion
        if (childSpanId is int) {
```

File: README.md [hunk 6]
```diff
@@ -163,107 +160,129 @@ service<http:Service> StudentData bind studentServiceListener {
...
+        http:Response response = new;
 
-        errors++;
-        io:println(errors);
+        studentData.errors += 1;
+        io:println(studentData.errors);
         // The below function adds tags that are to be passed as metrics in the traces. These tags are added to the default ootb system span.
-        _ = observe:addTagToSpan("error_counts", <string>errors);
-        _ = observe:addTagToSpan("tot_requests", <string>requestCounts);
+       _ = observe:addTagToSpan("tot_requests", <string>studentData.requestCounts);
```
Comment (Anoukh) [inline README.md]:
```suggestion
        _ = observe:addTagToSpan("tot_requests", <string>studentData.requestCounts);
```

File: README.md [hunk 7]
```diff
@@ -272,151 +291,157 @@ service<http:Service> StudentData bind studentServiceListener {
...
-        //  The below function adds tags that are to be passed as metrics in the traces. These tags are added to the default ootb system span.
-        _ = observe:addTagToSpan("tot_requests", <string>requestCounts);
-        _ = observe:addTagToSpan("error_counts", <string>errors);
-    }
+     @http:ResourceConfig {
+         methods: ["GET"],
+         path: "/getMarks/{stuId}"
+     }
+     // Get marks resource for obtaining marks of a particular student.
+     resource function getMarks(http:Caller httpConnection, http:Request request, int stuId) {
```
Comment (Anoukh) [inline README.md]: Shall we rename stuId to studentId?

File: README.md [hunk 8]
```diff
@@ -272,151 +291,157 @@ service<http:Service> StudentData bind studentServiceListener {
...
+         studentData.requestCounts += 1;
+         http:Response response = new;
+         json result ={};
```
Comment (Anoukh) [inline README.md]:
```suggestion
         json result = {};
```

File: README.md [hunk 9]
```diff
@@ -272,151 +291,157 @@ service<http:Service> StudentData bind studentServiceListener {
...
+
+         // Self defined span for observability purposes.
+         int|error firstSpan =  observe:startSpan("First span");
+         // Request made for obtaining marks of the student with the respective stuId to marks Service.
+         var requestReturn = marksServiceEP->get("/marks/getMarks/" + untaint stuId);
+
+         if (requestReturn is error) {
+
```
Comment (Anoukh) [inline README.md]: Shall we log the error?

File: README.md [hunk 10]
```diff
@@ -272,151 +291,157 @@ service<http:Service> StudentData bind studentServiceListener {
...
+             if (msg is error) {
+                 log:printError("Error", err = msg);
+             }
+             else {
+                 result = msg;
+             }
+         }
+
+         // Stopping the previously started span.
+         if (firstSpan is int ) {
```
Comment (Anoukh) [inline README.md]:
```suggestion
         if (firstSpan is int) {
```

File: README.md [hunk 11]
```diff
@@ -272,151 +291,157 @@ service<http:Service> StudentData bind studentServiceListener {
...
-  # + return - This function returns a JSON object. If data is added it returns JSON containing a status and id of student added.
-  #          If data is not added , it returns the JSON containing a status and error message.
+# `insertData()` is a function to add data to student records database.
+#
+# + name - This is the name of the student to be added.
+# + age -Student age.
+# + mobNo -Student mobile number.
+# + address - Student address.
+# + return - This function returns a JSON object. If data is added it returns JSON containing a status and id of student
+#            added.
```
Comment (Anoukh) [inline README.md]: Formatting issue?

File: README.md [hunk 12]
```diff
@@ -272,151 +291,157 @@ service<http:Service> StudentData bind studentServiceListener {
...
+#          If data is not added , it returns the JSON containing a status and error message.
 
 public function insertData(string name, int age, int mobNo, string address) returns (json) {
-    json updateStatus;
-    int uid;
+    json updateStatus = { "Status": "Data Inserted "};
```
Comment (Anoukh) [inline README.md]:
```suggestion
    json updateStatus = { "Status": "Data Inserted " };
```

File: README.md [hunk 13]
```diff
@@ -272,151 +291,157 @@ service<http:Service> StudentData bind studentServiceListener {
...
-        error e => io:println("Select data from student table failed: " + e.message);
-    }
-    return dataTable;
-}
 
+    string sqlString = "SELECT * FROM student WHERE mobNo = ?";
+    // Retrieve student data by invoking select remote function defined in ballerina sql client
+    var ret = studentDB->select(sqlString, Student, mobNo);
+
+        table< Student > dataTable = table{};
```
Comment (Anoukh) [inline README.md]: Formatting issue? Should be indented to the left as well

File: README.md [hunk 14]
```diff
@@ -545,60 +574,70 @@ import ballerina/http;
...
+            }
+
+            var intOp = int.convert(choice);
+
+            if (intOp is int){
+                operation = intOp;
+            }
+
+            // Program runs until the user inputs 6 to terminate the process.
+            if (operation == 6) {
```
Comment (Anoukh) [inline README.md]: Can we use the match statment here? Refer https://ballerina.io/learn/by-example/match.html

File: README.md [hunk 15]
```diff
@@ -608,148 +647,136 @@ function addStudent(http:Request req) {
     var mobile = io:readln("Enter mobile number: ");
     var add = io:readln("Enter Student address: ");
 
-    // Create the request as JSON message.
-    json jsonMsg = { "name": name, "age": check <int>age, "mobNo": check <int>mobile, "address": add };
-    req.setJsonPayload(jsonMsg);
+    var intAge =  int.convert(age);
+    var intMob = int.convert(mobile);
```
Comment (Anoukh) [inline README.md]: Maybe use a meaningful name like `mobileNumberAsInt`. Check in other places

File: README.md [hunk 16]
```diff
@@ -163,107 +160,129 @@ service<http:Service> StudentData bind studentServiceListener {
...
         methods: ["GET"],
         path: "/deleteStu/{stuId}"
     }
+
     // Delete Students resource for deleteing a student using id.
-    deleteStudent(endpoint httpConnection, http:Request request, int stuId) {
-        requestCounts++;
-        http:Response response;
+    resource function deleteStudent(http:Caller httpConnection, http:Request request, int stuId) {
+         studentData.requestCounts += 1;
```
Comment (Bhashinee) [inline README.md]:
```suggestion
        studentData.requestCounts += 1;
```

File: README.md [hunk 17]
```diff
@@ -608,148 +647,136 @@ function addStudent(http:Request req) {
...
+    if (response is http:Response ){
+        var jsonMsg = response.getJsonPayload();
+
+        if(jsonMsg is json){
+            string message = "";
+
+            if(jsonMsg.length() >=1){
+                int i = 0;
+                while (i < jsonMsg.length()) {
+                message = "Student Name: " + jsonMsg[i]["name"] .toString() + ", " + " Student Age: " + jsonMsg[i]["age"] .toString();
```
Comment (Anoukh) [inline README.md]: Formatting issues with indentation

File: README.md [hunk 18]
```diff
@@ -163,107 +160,129 @@ service<http:Service> StudentData bind studentServiceListener {
...
-            json jsonResult => {
-                status = jsonResult;
-            }
-            error e => {
-                status = { "Status": "Data Not available", "Error": e.message };
-            }
+        var jsonConversionValue = json.convert(dataTable);
+        if (jsonConversionValue is error) {
+            status = { "Status": "Data Not available" };
+        }
```
Comment (Bhashinee) [inline README.md]:
```suggestion
        } else {
```

File: README.md [hunk 19]
```diff
@@ -163,107 +160,129 @@ service<http:Service> StudentData bind studentServiceListener {
...
-        _ = httpConnection->respond(response) but { error e => log:printError("Error sending response", err = e) };
+        var result = httpConnection->respond(response);
+        if (result is error) {
+            log:printError("Error sending response", err = result);
+        }
 
-        _ = observe:finishSpan(childSpanId);
+        if (childSpanId is int ) {
+            _ = observe:finishSpan(childSpanId);
+        }
```
Comment (Bhashinee) [inline README.md]:
```suggestion
        } else {
```

Comment (Bhashinee) [reply README.md]: Fix in other places as well.

File: README.md [hunk 20]
```diff
@@ -449,34 +474,38 @@ import ballerina/mysql;
 import ballerina/observe;
 import ballerina/runtime;
 
+
```
Comment (Bhashinee) [inline README.md]: Shall we remove this extra new line?

File: README.md [hunk 21]
```diff
@@ -491,28 +520,28 @@ public function findMarks(int stuId) returns (json) {
...
-        table tableReturned => dataTable = tableReturned;
-        error err => {
-            log:printError(err.message,err= err);
-            status = { "Status": "Select data from student table failed: ", "Error": err.message };
-            return status;
-        }
+    table<Marks> dataTable = table {};
+
+    if (returnValue is table<Marks>){
+            dataTable = returnValue;
```
Comment (Bhashinee) [inline README.md]:
```suggestion
        dataTable = returnValue;
```

File: README.md [hunk 22]
```diff
@@ -491,28 +520,28 @@ public function findMarks(int stuId) returns (json) {
...
-    match returnValue {
-        table tableReturned => dataTable = tableReturned;
-        error err => {
-            log:printError(err.message,err= err);
-            status = { "Status": "Select data from student table failed: ", "Error": err.message };
-            return status;
-        }
+    table<Marks> dataTable = table {};
+
+    if (returnValue is table<Marks>){
```
Comment (Bhashinee) [inline README.md]:
```suggestion
    if (returnValue is table<Marks>) {
```

File: README.md [hunk 23]
```diff
@@ -491,28 +520,28 @@ public function findMarks(int stuId) returns (json) {
...
+            dataTable = returnValue;
+    }
+    else {
+        log:printError("Error Detected",err= returnValue);
```
Comment (Bhashinee) [inline README.md]:
```suggestion
        log:printError("Error Detected", err = returnValue);
```

File: README.md [hunk 24]
```diff
@@ -491,28 +520,28 @@ public function findMarks(int stuId) returns (json) {
...
-            status = { "Status": "Data Not available", "Error": err.message };
-            log:printError(err.message,err = err);
-        }
+    var jsonConversionValue = json.convert(dataTable);
+
+    if (jsonConversionValue is json){
+        status = jsonConversionValue;
+    }
+    else {
+        status = { "Status": "Data Not available"};
```
Comment (Bhashinee) [inline README.md]:
```suggestion
        status = { "Status": "Data Not available" };
```

File: README.md [hunk 25]
```diff
@@ -491,28 +520,28 @@ public function findMarks(int stuId) returns (json) {
...
+        log:printError("Error Detected",err = jsonConversionValue);
```
Comment (Bhashinee) [inline README.md]:
```suggestion
        log:printError("Error Detected", err = jsonConversionValue);
```

File: README.md [hunk 26]
```diff
@@ -608,148 +647,136 @@ function addStudent(http:Request req) {
     var mobile = io:readln("Enter mobile number: ");
     var add = io:readln("Enter Student address: ");
 
-    // Create the request as JSON message.
-    json jsonMsg = { "name": name, "age": check <int>age, "mobNo": check <int>mobile, "address": add };
-    req.setJsonPayload(jsonMsg);
+    var intAge =  int.convert(age);
```
Comment (Bhashinee) [inline README.md]:
```suggestion
    var intAge = int.convert(age);
```

File: README.md [hunk 27]
```diff
@@ -608,148 +647,136 @@ function addStudent(http:Request req) {
...
-            var msg = response.getJsonPayload();
-            //obtaining the result from the response received
-            match msg {
-                json jsonPL => {
-                    string message = "Status: " + jsonPL["Status"] .toString() + " Added Student Id :- " +
-                        jsonPL["id"].toString();
-                    // Extracting data from JSON received and displaying.
-                    io:println(message);
-                }
+    if(intAge is int && intMob is int) {
```
Comment (Bhashinee) [inline README.md]:
```suggestion
    if (intAge is int && intMob is int) {
```

File: README.md [hunk 28]
```diff
@@ -608,148 +647,136 @@ function addStudent(http:Request req) {
...
+        // Create the request as JSON message.
+        json jsonMsg = { "name": name, "age":intAge, "mobNo":intMob, "address": add ,"id": 0};
```
Comment (Bhashinee) [inline README.md]:
```suggestion
        json jsonMsg = { "name": name, "age":intAge, "mobNo":intMob, "address": add ,"id": 0 };
```

File: README.md (hunk 19 above)
Comment (Shairam) [reply README.md]: Yes i will make the necessary changes.

File: README.md (hunk 14 above)
Comment (Shairam) [reply README.md]: Yes I will make the necessary changes.

File: README.md [hunk 29]
```diff
@@ -163,261 +160,279 @@ service<http:Service> StudentData bind studentServiceListener {
...
-                        log:printError(er.message, err = er);
-                    }
-                }
+        int|error firstSpan = observe:startSpan("First span");
+        // Request made for obtaining marks of the student with the respective studentId to marks Service.
+        var requestReturn = marksServiceEP->get("/marks/getMarks/" + untaint studentId);
+
+        if (requestReturn is error) {
+            log:printError("Error", err = requestReturn);
+        }
```
Comment (Bhashinee) [inline README.md]:
```suggestion
        } else {
```
Fix all the if else blocks

Comment (Shairam) [reply README.md]: Yes I will fix them

File: guide/client_service/client_main.bal [hunk 30]
```diff
@@ -42,184 +40,174 @@ public function main() {
...
-                        // Notify user if no records are available.
-                        message = "\n Student record is empty";
-                        io:println(message);
-                    }
-                }
-                error err => {
-                    log:printError(err.message, err = err);
+    var response = studentService->post("/records/viewAll", null);
+
+    if (response is http:Response ){
```
Comment (Bhashinee) [inline guide/client_service/client_main.bal]:
```suggestion
    if (response is http:Response) {
```

File: guide/client_service/client_main.bal [hunk 31]
```diff
@@ -42,184 +40,174 @@ public function main() {
...
+        var jsonMsg = response.getJsonPayload();
+
+        if(jsonMsg is json){
```
Comment (Bhashinee) [inline guide/client_service/client_main.bal]:
```suggestion
        if (jsonMsg is json) {
```

File: guide/client_service/client_main.bal [hunk 32]
```diff
@@ -42,184 +40,174 @@ public function main() {
...
+            string message = "";
+
+            if(jsonMsg.length() >=1){
```
Comment (Bhashinee) [inline guide/client_service/client_main.bal]:
```suggestion
            if (jsonMsg.length() >=1) {
```

File: guide/client_service/client_main.bal [hunk 33]
```diff
@@ -42,184 +40,174 @@ public function main() {
...
+                int i = 0;
+                while (i < jsonMsg.length()) {
+
+                message = "Student Name: " + jsonMsg[i]["name"] .toString() + ", " + " Student Age: " +
+                jsonMsg[i]["age"] .toString();
```
Comment (Bhashinee) [inline guide/client_service/client_main.bal]:
```suggestion
                jsonMsg[i]["age"].toString();
```

File: guide/client_service/client_main.bal (hunk 31 above)
Comment (Shairam) [reply guide/client_service/client_main.bal]: Fixed formatting issues

File: README.md [hunk 34]
```diff
@@ -129,31 +129,28 @@ type Student record {
 };
 
 // Endpoint for marks details client.
-endpoint http:Client marksServiceEP {
-    url: " http://localhost:9191"
-};
+http:Client marksServiceEP = new("http://localhost:9191",
+    config = { httpVersion: "2.0" });
```
[the remark this answers is not in the document]
Comment (Shairam) [reply README.md]: Yeah sure I will do it.

File: README.md [hunk 35]
```diff
@@ -163,261 +158,274 @@ service<http:Service> StudentData bind studentServiceListener {
...
-    match returnValue {
-        table tableReturned => dataTable = tableReturned;
-        error e => io:println("Select data from student table failed: " + e.message);
+
+    string sqlString = "SELECT * FROM student WHERE mobNo = ?";
+    // Retrieve student data by invoking select remote function defined in ballerina sql client
+    var ret = studentDB->select(sqlString, Student, mobNo);
+
+    table<Student> dataTable = table{};
+    if (ret is error) {
```
Comment (Bhashinee) [inline README.md]: Shall we handle this error? At least log the error?

Comment (Shairam) [reply README.md]: Yes, I will update it.

File: guide/students/student_management_service.bal [hunk 36]
```diff
@@ -65,259 +61,269 @@ service<http:Service> StudentData bind studentServiceListener {
...
+    var ret = studentDB->select(sqlString, Student, mobNo);
 
+    table<Student> dataTable = table{};
+    if (ret is error) {
+        log:printError("Error ", err = ret);
+    } else {
+        dataTable = ret;
+    }
+    return dataTable;
+}
```
Comment (Anoukh) [inline guide/students/student_management_service.bal]:
```suggestion
}

```

File: guide/students/tests/students_service_test.bal [hunk 37]
```diff
@@ -13,17 +13,28 @@ function testingMockError() {
     // Initialize the empty HTTP request.
     http:Request request;
     // Send 'GET' request and obtain the response.
-    http:Response response = check studentData->get("/records/testError");
-    string res = check  response.getTextPayload();
-    test:assertEquals(res,"Test Error made", msg = "Test error success");
+    var response = studentService->get("/records/testError");
+    if (response is http:Response){
```
Comment (Anoukh) [inline guide/students/tests/students_service_test.bal]:
```suggestion
    if (response is http:Response) {
```

File: guide/students/tests/students_service_test.bal [hunk 38]
```diff
@@ -13,17 +13,28 @@ function testingMockError() {
...
 @test:Config
 function invalidDataDeletion() {
     http:Request request;
     // Send 'GET' request and obtain the response.
-    http:Response response = check studentData->get("/records/deleteStu/9999");
-    // Expected response JSON is as below.
-    json res = check  response.getJsonPayload();
-    test:assertEquals(res.toString(),"{\"Status\":\"Data Not Found\"}",msg = "Test error success");
+    var response =  studentService->get("/records/deleteStu/9999");
+    if (response is http:Response){
```
Comment (Anoukh) [inline guide/students/tests/students_service_test.bal]:
```suggestion
    if (response is http:Response) {
```

File: README.md [hunk 39]
```diff
@@ -163,261 +158,275 @@ service<http:Service> StudentData bind studentServiceListener {
...
 
-        // Send the response back to the client with the returned JSON value from insertData function.
-        response.setJsonPayload(returnValue);
-        _ = httpConnection->respond(response) but { error e => log:printError("Error sending response", err = e)};
+            if (studentDetails is Student) {
+                io:println(studentDetails);
+                // Calling the function insertData to update database.
+                json returnValue = insertData(untaint studentDetails.name, untaint studentDetails.age, untaint
+                    studentDetails.mobNo, untaint studentDetails.
+                    address);
```
Comment (Anoukh) [inline README.md]: This can be taken to the top line

File: README.md [hunk 40]
```diff
@@ -163,261 +158,275 @@ service<http:Service> StudentData bind studentServiceListener {
...
-        table tableReturned => dataTable = tableReturned;
-        error e => io:println("Select data from student table failed: " + e.message);
+
+    string sqlString = "SELECT * FROM student WHERE mobNo = ?";
+    // Retrieve student data by invoking select remote function defined in ballerina sql client
+    var ret = studentDB->select(sqlString, Student, mobNo);
+
+    table<Student> dataTable = table{};
+    if (ret is error) {
+    log:printError("Error ", err = ret);
```
Comment (Anoukh) [inline README.md]:
```suggestion
        log:printError("Error ", err = ret);
```

**Review by Anoukh — APPROVED** [review]

**Pull request merged**